tools/pd-ctl/pdctl/ctl.go

```diff
@@ -68,11 +87,55 @@ func Start(args []string) {
 		command.NewComponentConfigCommand(),
 	)
 
-	rootCmd.SetArgs(args)
+	rootCmd.Flags().ParseErrorsWhitelist.UnknownFlags = true
 	rootCmd.SilenceErrors = true
+
+	return rootCmd
+}
+
+func getRootCmd(args []string) *cobra.Command {
+	rootCmd := getBasicCmd()
+
+	rootCmd.SetArgs(args)
+	rootCmd.ParseFlags(args)
+	rootCmd.SetOutput(os.Stdout)
+
+	return rootCmd
+}
+
+func getMainCmd(args []string) *cobra.Command {
+	rootCmd := getBasicCmd()
+
+	rootCmd.Flags().BoolVarP(&detach, "detach", "d", true, "Run pdctl without readline.")
+	rootCmd.Flags().BoolVarP(&interact, "interact", "i", false, "Run pdctl with readline.")
+	rootCmd.Flags().BoolVarP(&version, "version", "V", false, "Print version information and exit.")
+	rootCmd.Run = pdctlRun
+
+	rootCmd.SetArgs(args)
 	rootCmd.ParseFlags(args)
 	rootCmd.SetOutput(os.Stdout)
 
+	return rootCmd
+}
+
+// MainStart start main command
+func MainStart(args []string) {
```

Comment:
Looks like these two functions are almost the same except the `rootCmd`, can we combine them?

Reply:
I refactor the code, PTAL.